src/simplewallet/simplewallet.cpp

```diff
@@ -119,8 +125,9 @@ using tools::fee_priority;
 #define SCOPED_WALLET_UNLOCK_ON_BAD_PASSWORD(code) \
   LOCK_IDLE_SCOPE(); \
   boost::optional<tools::password_container> pwd_container = boost::none; \
-  if (m_wallet->ask_password() && !(pwd_container = get_and_verify_password())) { code; } \
-  tools::wallet_keys_unlocker unlocker(*m_wallet, pwd_container ? &pwd_container->password() : nullptr);
+  if (m_wallet_impl->getAskPasswordType() && !(pwd_container = get_and_verify_password())) { code; } \
```

Comment:
A small stylistic thing: I would compare against `AskPassword_Never` instead of relying on the fact that this enum value is 0.